cli/src/utils.rs

```diff
@@ -254,39 +255,42 @@ pub fn enable_paint() {
 
 /// Gives out a provider with a `100ms` interval poll if it's a localhost URL (most likely an anvil
 /// node) and with the default, `7s` if otherwise.
-pub fn get_http_provider(url: &str) -> Provider<Http> {
-    let provider = Provider::try_from(url).expect("Bad fork provider.");
+pub fn get_http_provider(url: &str, aggressive: bool) -> Arc<Provider<RetryClient<Http>>> {
+    let (max_retry, initial_backoff) = if aggressive { (1000, 1) } else { (10, 1000) };
+    let interval = if aggressive { Duration::from_secs(1) } else { Duration::from_secs(7) };
 
-    if url.contains("127.0.0.1") || url.contains("localhost") {
+    let provider = Provider::<RetryClient<Http>>::new_client(url, max_retry, initial_backoff)
+        .expect("Bad fork provider.");
+
+    Arc::new(if url.contains("127.0.0.1") || url.contains("localhost") {
         provider.interval(Duration::from_millis(100))
     } else {
-        provider
-    }
+        provider.interval(interval)
+    })
 }
 
-pub fn print_receipt(receipt: &TransactionReceipt, nonce: U256) -> eyre::Result<()> {
+pub fn print_receipt(receipt: &TransactionReceipt) -> eyre::Result<()> {
     let mut contract_address = "".to_string();
     if let Some(addr) = receipt.contract_address {
         contract_address = format!("\nContract Address: 0x{}", hex::encode(addr.as_bytes()));
     }
 
     let gas_used = receipt.gas_used.unwrap_or_default();
-    let gas_price = receipt.effective_gas_price.expect("no gas price");
+    let gas_price = receipt.effective_gas_price.unwrap_or_default();
     let paid = format_units(gas_used.mul(gas_price), 18)?;
```

Comment:
since the gas price is coming from the receipt, I think this is safe to and we should consider expect("") here which would get rid of the Result

Reply:
Actually, I found out that Optimism receipts do not come with a `gas_price`, and was panicking here. 

I'd rather not panic at all when printing receipts, and just show on the terminal invalid information. The actual receipt is always saved correctly in the broadcast log anyway.

Reply:
hmm that's weird, 

`curl https://kovan.optimism.io -X POST -H "Content-Type: application/json" -d'{"jsonrpc":"2.0","method":"eth_getTransactionByHash","params":["0x8ab5308d870749ebc10da8d0f52b28a3be506cdb8c51f8a96f8fff3818c1803c"],"id":0}' | jq`

includes it.

nevertheless good, point, should we therefore print the `Paid` conditionally or rather make `Paid: N/A`?